src/librustc/util/common.rs

```diff
@@ -29,6 +33,59 @@ pub struct ErrorReported;
 
 thread_local!(static TIME_DEPTH: Cell<usize> = Cell::new(0));
 
+/// Initialized for -Z profile-queries
+thread_local!(static PROFQ_CHAN: RefCell<Option<Sender<ProfileQueriesMsg>>> = RefCell::new(None));
```

Comment:
Ah, that explains a bunch. If I get a satisfactory explanation out of him this may be good to go.